### PR TITLE
#161967131 User should be able to rate article and view article rating

### PR DESCRIPTION
**Pull request by Darius-Ndubi**
## What does this PR do?
 Allow authenticated users to rate an article.
 Allow any user to view an article's rating.

## Description of tasks to be completed  
 - Add feature to allow *only* authenticated users to rate an article and edit their previous rating.
 - Add feature to retrieve the rating of a particular article.
 - Add tests on the features above.

## How should this be manually tested
 - Clone the repo from your terminal: `git clone https://github.com/andela/ah-code-titans.git`
 - Create a virtual environment:`python3 -m venv <name_of_environment>`
 - Activate the virtual environment:  `source <name_of_environment>/bin/activate`
 - Install project dependencies: `pip install -r requirements.txt`
- Add the following environment variables and populate them with your settings:
  - `export DB_NAME='choose_a_dbname'`
  - `export DB_USER='db_username'`
  - `export DB_PASSWORD='db_role_password'`
  - `export DB_HOST='localhost or preffered host'`
  - `export DB_PORT='server_port_number'`
  - `export SECRET_KEY='your_secret_key'`
  - `export EMAIL_HOST_USER="codetitans32@gmail.com"`
  - `export EMAIL_HOST_PASSWORD='Codetitans@32'`
- Switch to branch `ft-rate-article-161967131`
- Run migrations by:
    -  `python manage.py makemigrations`
    - `python manage.py migrate`
- Run tests by:
    - `python manage.py test`

## What are the relevant Pivotal tracker stories
[#161967131](https://www.pivotaltracker.com/story/show/161967131)

## Checklist

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation

- [x] My changes generate no new warnings

## Rate an article
screenshot 2018-12-13 at 18 11 37

## Get article rating
screenshot 2018-12-13 at 18 11 48

### CODE REVIEW

File: authors/apps/rating/tests/test_rate_get.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            activate_url,
+            content_type='application/json'
+        )
+        test_token = self.login(self.activated_user)
+        global user_token
+        user_token = test_token.data['token']
+
+        return test_token.data['token']
+
+    def email_verification_and_login_user2(self):
```
Comment (mutoromike) [inline authors/apps/rating/tests/test_rate_get.py]: Consider refactoring these three methods. They are in two files.

**Review by mutoromike — CHANGES_REQUESTED** [review]
Consider refactoring the methods used in the tests. You can have them in one file instead of them repeating.

File: authors/apps/rating/tests/test_rate_get.py (hunk 1 above)
Comment (Darius-Ndubi) [reply authors/apps/rating/tests/test_rate_get.py]: okay @mutoromike  let me work on that

Comment (Darius-Ndubi) [reply authors/apps/rating/tests/test_rate_get.py]: @mutoromike the changes have been made. thank you.

**Review by denomash — APPROVED** [review]
👍

**Review by davishooly — APPROVED** [review]

**Review by mutoromike — APPROVED** [review]

**Pull request merged**